app/services/channels/approve_channel_transfer.rb

```diff
@@ -0,0 +1,44 @@
+module Channels
+  class ApproveChannelTransfer < BaseService
+
+    def initialize(channel:, should_delete: true)
+      @channel = channel
+      @should_delete = should_delete
+    end
+
+    def perform
+      # Verify the contested_by channel
+      contested_by = @channel.contested_by_channel
```

Comment:
We should lock this in a https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html